### DISCUSSION TITLE
Copilot and Clojure parentheses

### DISCUSSION THREAD

**Discussion by btsheehy** [community/community#16261] [Copilot Conversations]
While Copilot often produces helpful suggestions for my Clojure code, it never seems to take into account closing parentheses or brackets that my editor has already added in. For example, if I type
```clojure
(defn func [a] (let [b (-> a :c str
```
my editor will have already added in `)]))`. But when Copilot makes its suggestion, it will often include many of the same closing characters. This makes my code a mess anytime I accept a suggestion and I spend just as much time hunting down mismatching brackets as I would have if I'd just written the code myself.

**Comment by mode89** [comment]
Same here. Using copilot.vim.

**Comment by rphuber** [comment]
I'm not sure it's just an issue with closure.  It seems to be an issue with delimiters in vim that exist when you complete.  Copilot vim inserts them even though they already exist.  Here is another discussion describing it in other languages https://github.com/orgs/community/discussions/46277.

**Comment by FhyTan** [comment]
The same issue exists in VSCode. Here’s a snapshot of Copilot prompting a suggestion with parentheses that the code already has.
![image](https://user-images.githubusercontent.com/16273230/236595482-e87736ef-6d54-411c-970f-d8544f846ef3.png)
